ffi/src/expressions.rs

```diff
@@ -245,3 +249,417 @@ pub extern "C" fn visit_expression_literal_bool(
 ) -> usize {
     wrap_expression(state, Expression::literal(value))
 }
+
+#[handle_descriptor(target=Expression, mutable=false, sized=true)]
+pub struct SharedExpression;
+
+/// Free the memory the passed SharedExpression
+///
+/// # Safety
+/// Engine is responsible for passing a valid SharedExpression
+#[no_mangle]
+pub unsafe extern "C" fn free_kernel_predicate(data: Handle<SharedExpression>) {
+    data.drop_handle();
+}
+
+/// Constructs a kernel expression that is passed back as a SharedExpression handle
+///
+/// # Safety
+/// The caller is responsible for freeing the retured memory, either by calling
+/// [`free_kernel_predicate`], or [`Handle::drop_handle`]
+#[no_mangle]
+pub unsafe extern "C" fn get_kernel_expression() -> Handle<SharedExpression> {
```

Comment:
I currently have this C function that returns a big expression for testing.  How can I limit visibility while still passing this over to something like `read_table`?